src/devnet/docker-devnet.ts

```diff
@@ -0,0 +1,50 @@
+import { HardhatDocker, Image } from "@nomiclabs/hardhat-docker";
+import { ChildProcess, spawn, spawnSync } from "child_process";
+import { DEVNET_DOCKER_REPOSITORY } from "../constants";
+
+import { IntegratedDevnet } from "./integrated-devnet";
+
+const CONTAINER_NAME = "integrated-devnet" as const;
+
+export class DockerDevnet extends IntegratedDevnet {
+    private docker: HardhatDocker;
+
+    constructor(private image: Image, host: string, port: string) {
+        super(host, port);
+    }
+
+    private async pullImage() {
+        if (!this.docker) {
+            this.docker = await HardhatDocker.create();
+        }
+
+        if (!(await this.docker.hasPulledImage(this.image))) {
+            console.log(`Pulling image ${HardhatDocker.imageToRepoTag(this.image)}`);
+            await this.docker.pullImage(this.image);
+        }
+    }
+
+    protected async spawnChildProcess(): Promise<ChildProcess> {
+        await this.pullImage();
+
+        console.log(`Starting the "${CONTAINER_NAME}" Docker container`);
+
+        return spawn("docker", [
+            "run",
+            "--detach",
+            "--rm",
+            "--name",
+            CONTAINER_NAME,
+            "-it",
```

Comment:
Are the `-it` flags necessary?

Reply:
[a075157](https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/77/commits/a0751578814f7bb71e9bbadc8117a63a505c1e73)